src/components/PanelMenuWrapper.js

```diff
@@ -53,16 +56,28 @@ class PanelsWithSidebar extends Component {
       group = child.props.group;
       name = child.props.name;
 
-      if (groupLookup.hasOwnProperty(group)) {
-        groupIndex = groupLookup[group];
-        obj = groups[groupIndex];
-      } else {
+      if (child.type.name === 'Button') {
+        if (groupLookup.hasOwnProperty(group)) {
+          throw new Error('cannot have 2 menu buttons with same name');
+        }
         groupLookup[group] = groups.length;
-        obj = {name: group, panels: []};
+        obj = {
+          name: group,
+          type: 'Button',
+          component: child,
+        };
         groups.push(obj);
+      } else {
+        if (groupLookup.hasOwnProperty(group)) {
+          groupIndex = groupLookup[group];
+          obj = groups[groupIndex];
+        } else {
+          groupLookup[group] = groups.length;
+          obj = {name: group, panels: []};
```

Comment:
Could `obj` be `{name: group}` with no panels? Then later on we would handle the case where `panels` is undefined?

not a huge deal but having to create empty arrays seems to suggest we generalize the framework just a bit.